File: dev-portal/python/postman_sdk/types/mock.py
"""
Mock server-related types
"""

from pydantic import BaseModel, Field


class MockServer(BaseModel):
    """Mock server entity"""
    id: str
    uid: str
    name: str
    owner: str | None = None
    collection: str
    environment: str | None = None
    mock_url: str = Field(alias="mockUrl")
    is_public: bool = Field(alias="isPublic", default=False)
    created_at: str | None = Field(None, alias="createdAt")
    updated_at: str | None = Field(None, alias="updatedAt")

    class Config:
        populate_by_name = True


class CreateMockRequest(BaseModel):
    """Create mock server request"""
    name: str
    collection: str
    workspace_id: str
    environment: str | None = None
    is_private: bool = False


class CreateMockResult(BaseModel):
    """Create mock server result"""
    success: bool
    mock: MockServer | None = None
    error: str | None = None


class MockMapping(BaseModel):
    """Mock server mapping"""
    mock_id: str
    mock_url: str
    name: str
    collection_name: str
    collection_uid: str


class MockUrlVariable(BaseModel):
    """Mock URL variable for environment"""
    key: str
    value: str
    enabled: bool = True
    type: str = "default"

File: dev-portal/typescript/src/types/mock.ts
/**
 * Mock server-related types
 */

/**
 * Mock server entity
 */
export interface MockServer {
  id: string;
  uid: string;
  name: string;
  owner?: string;
  collection: string;
  environment?: string;
  mockUrl: string;
  isPublic: boolean;
  createdAt?: string;
  updatedAt?: string;
  config?: MockServerConfig;
}

/**
 * Mock server configuration
 */
export interface MockServerConfig {
  serverResponseId?: string;
  matchBody?: boolean;
  matchQueryParams?: boolean;
  matchWildcards?: boolean;
  delay?: MockDelayConfig;
}

/**
 * Mock delay configuration
 */
export interface MockDelayConfig {
  type: 'fixed' | 'random';
  preset?: 'fast' | 'slow' | 'custom';
  duration?: number;
  min?: number;
  max?: number;
}

/**
 * Create mock server request
 */
export interface CreateMockRequest {
  name: string;
  collection: string;
  workspaceId: string;
  environment?: string;
  isPrivate?: boolean;
  config?: MockServerConfig;
}

/**
 * Create mock server result
 */
export interface CreateMockResult {
  success: boolean;
  mock?: MockServer;
  error?: string;
}

/**
 * Mock server mapping
 */
export interface MockMapping {
  mockId: string;
  mockUrl: string;
  name: string;
  collectionName: string;
  collectionUid: string;
}

/**
 * Mock URL variable for environment
 */
export interface MockUrlVariable {
  key: string;
  value: string;
  enabled: boolean;
  type: 'default';
}

/**
 * Mock call log entry
 */
export interface MockCallLog {
  id: string;
  mockId: string;
  timestamp: string;
  method: string;
  path: string;
  responseCode: number;
  responseTime: number;
}

/**
 * Delete mock result
 */
export interface DeleteMockResult {
  success: boolean;
  error?: string;
}
